scripts/update_proxy.js

```diff
@@ -0,0 +1,29 @@
+const { ethers } = require("hardhat");
+
+async function main() {
+    // Replace these with your actual addresses
+    const proxyAddress = "0xYourProxyContractAddress";
+    const newImplementationAddress = "0xYourNewImplementationContractAddress";
+
+    // Get the admin signer
+    const [admin] = await ethers.getSigners();
+
+    // ABI of the TransparentUpgradeableProxy
+    const ProxyAdminABI = [
+        "function upgrade(address proxy, address implementation) external"
```

Comment:
Hey @isa16alejandra, I'm not completely sure how to complete this one.

Reply:
@TheSlackOne This is the script I created to upgrade the contract.

```
    // Replace these with your actual addresses
    const proxyContractAddress = "0xYourProxyContractAddress";
    const proxyAdminContractAddress = "0xYourProxyAdminContractAddress";
    const implementationAddress = '0xYourNewImplementationContractAddress';


    // Create a contract instance for the proxy
    const proxyAdmin = await ethers.getContractAt(PROXY_ADMIN_ABI, proxyAdminContractAddress, proxyAdminOwnerSigner);
    const originalImplementation = await proxyAdmin.getProxyImplementation(proxyContractAddress);
    console.log(`Proxy ${proxyContractAddress} is currently pointing to implementation at: ${originalImplementation}`);

    // Execute the upgrade
    const tx = await proxyAdmin.upgrade(proxyContractAddress, implementationAddress);
    await tx.wait();

    console.log(`Proxy ${proxyContractAddress} upgraded to new implementation at: ${await proxyAdmin.getProxyImplementation(proxyContractAddress)}`);
```


```
    const PROXY_ADMIN_ABI = [{
        "inputs": [
            {
                "internalType": "contract TransparentUpgradeableProxy",
                "name": "proxy",
                "type": "address"
            },
            {
                "internalType": "address",
                "name": "implementation",
                "type": "address"
            }
        ],
        "name": "upgrade",
        "outputs": [],
        "stateMutability": "nonpayable",
        "type": "function"
    },
    {
        "inputs": [
            {
                "internalType": "contract TransparentUpgradeableProxy",
                "name": "proxy",
                "type": "address"
            }
        ],
        "name": "getProxyImplementation",
        "outputs": [
            {
                "internalType": "address",
                "name": "",
                "type": "address"
            }
        ],
        "stateMutability": "view",
        "type": "function"
    }];
```